packages/next-swc/crates/core/src/disallow_amp_config_in_app_dir.rs

```diff
@@ -0,0 +1,70 @@
+use std::path::PathBuf;
+
+use next_binding::swc::core::{
+    common::{errors::HANDLER, FileName},
+    ecma::ast::*,
+    ecma::visit::{as_folder, noop_visit_mut_type, Fold, VisitMut},
+};
+pub fn disallow_amp_config_in_app_dir(
+    filepath: FileName,
+    app_dir: PathBuf,
+) -> impl Fold + VisitMut {
+    as_folder(DisallowAmpConfigInAppDir {
+        filepath: filepath.to_string(),
+        app_dir,
+    })
+}
+
+#[derive(Debug, Default)]
+struct DisallowAmpConfigInAppDir {
+    filepath: String,
+    app_dir: PathBuf,
+}
+
+const CONFIG_KEY: &str = "config";
+
+impl VisitMut for DisallowAmpConfigInAppDir {
+    noop_visit_mut_type!();
+
+    fn visit_mut_export_decl(&mut self, export: &mut ExportDecl) {
+        if let Decl::Var(var_decl) = &export.decl {
+            for decl in &var_decl.decls {
+                let is_config =
+                    matches!(&decl.name, Pat::Ident(ident) if &ident.id.sym == CONFIG_KEY);
+
+                if is_config {
```

Comment:
Could this just be a runtime check we do in dev inside of `next-app-loader` or `app-render`?

Reply:
Or potentially this should be a part of the TypeScript plugin cc @shuding

Reply:
Yes we can have both TS plugin + runtime check. Also with our new upcoming metadata field, we will bring many things to do runtime checking.